coverage/PENDING_REVIEW/intel/0449-COVERAGE-nir-opt-if-199/cov-loop-returns-behind-true-and-false.amber

```diff
@@ -0,0 +1,228 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+# #define _int_1 _GLF_uniform_int_values[0]
+# #define _int_0 _GLF_uniform_int_values[1]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_int_values: [1, 0]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int _GLF_uniform_int_values[2];
+# };
+#
+# const int _GLF_global_loop_bound = 100;
+# int _GLF_global_loop_count = 0;
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     switch(_int_1)
+#     {
+#         case 0:
+#             if (true)
+#             {
+#                 _GLF_color = vec4(_int_0);
+#                 return;
+#             }
+#         case 1:
+#             if (true)
+#             {
+#                 _GLF_color = vec4(_int_1, _int_0, _int_0, _int_1);
+#
+#                 if (false)
```

Comment:
Would it be possible to change some/all of the "true" and "false"s to be opaque, e.g. encoding true as "_int_1 > _int_0" and false as "_int_1 < _int_0", while still hitting the same coverage point?

I don't think the tool for extracting literals into uniforms will do this, and it might be good to make the control flow less trivial for a compiler to analyse.

Reply:
We can do that only in case 0 that is never accessed. In other places changing true or false to anything else didn't work.